File: mono/nn/GnuplotRenderer.cs
using System;

namespace nn
{
	public class GnuplotRenderer<T,U> where T : FeatureRepresentation where U : Edge<T>
	{
		public static void outNodes(string basename, int steps, System.Collections.Generic.ICollection<T> features) 
		{
			// encodes the output as text.
			using (System.IO.StreamWriter file = 
			       new System.IO.StreamWriter(System.IO.Path.GetTempPath()+basename+" "+steps+".dat"))
			{
				foreach (FeatureRepresentation feature in features)
				{
					file.WriteLine(feature);
				}
			}
		}
		public static void outEdges(string basename, int steps, System.Collections.Generic.ICollection<U> edges) 
		{
			// encodes the output as text.
			using (System.IO.StreamWriter file = 
			       new System.IO.StreamWriter(System.IO.Path.GetTempPath()+basename+" "+steps+".dat"))
			{
				foreach (Edge<T> edge in edges)
				{
					file.Write(edge.A);
					file.Write(edge.B);
					file.WriteLine();
				}
			}
		}
	}
}

File: java/nn_multi_lang/src/main/java/de/elbosso/scratch/algorithms/nn/GnuplotRenderer.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package de.elbosso.scratch.algorithms.nn;

import java.io.IOException;

/**
 *
 * @author elbosso
 */
public class GnuplotRenderer
{
	static <T extends de.elbosso.algorithms.nn.FeatureRepresentation> void outNodes(java.lang.String basename, int steps, java.util.Collection<T> features) throws IOException
	{
		java.io.File f=java.io.File.createTempFile("aaaa", "bbbb");
		java.io.File dir=f.getParentFile();
		f.deleteOnExit();
		f=new java.io.File(dir,basename+" "+steps+".dat");
		java.io.PrintWriter pw=new java.io.PrintWriter(f);
		for (de.elbosso.algorithms.nn.FeatureRepresentation feature : features)
		{
			pw.println(feature);
		}
		pw.close();
	}
	static <T extends de.elbosso.algorithms.nn.Edge> void outEdges(java.lang.String basename, int steps, java.util.Collection<T> edges) throws IOException
	{
		java.io.File f=java.io.File.createTempFile("aaaa", "bbbb");
		java.io.File dir=f.getParentFile();
		f.deleteOnExit();
		f=new java.io.File(dir,basename+" "+steps+".dat");
		java.io.PrintWriter pw=new java.io.PrintWriter(f);
		for (de.elbosso.algorithms.nn.Edge edge : edges)
		{
			pw.println(edge.getA());
			pw.println(edge.getB());
			pw.println();
		}
		pw.close();
	}
}
